### PR TITLE
Add verify_auth endpoint

### PR DESCRIPTION
**Pull request by jdw6359**
I think this looks like what you want:
GET @ /api/verify_auth will route to sessions_controller#verify_auth
Method will attempt to lookup user based upon git hib magic (auth["provider"] and auth["uid"]), and return json true or false indicating auth.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -7,8 +12,20 @@ def create
...
   end
+
+  # verify the user is authenticated
+  def verify_auth
+    # attempt to access the user based on auth["provider"] and auth["uid"]
+    user = User.find_by_provider_and_uid(auth["provider"], auth["uid"])
+    if user
+      render json: {authenticated: true}
+    else
+      render json: {authenticated: true}
```
Comment (vuzDylan) [inline app/controllers/sessions_controller.rb]: I think this should be false

Comment (jdw6359) [inline app/controllers/sessions_controller.rb]: Lololol

**Comment by kocsenc** [comment]
#noob

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (kocsenc) [inline app/controllers/sessions_controller.rb]: The copy paste code syndrome.

Comment (kocsenc) [inline app/controllers/sessions_controller.rb]: Also, I think it'd be nice to use a simpler `key`. Something like `auth` or `authed`.

Comment (mok4ry) [inline app/controllers/sessions_controller.rb]: I disagree. `auth` being true seems more like you're saying "this requires authentication," and while `authed` conveys "you've been authenticated" it's still not immediately obvious.

Comment (kocsenc) [inline app/controllers/sessions_controller.rb]: :+1:  Never hurts to be verbose.

**Pull request merged**